lib/Ast.ml

```diff
@@ -1054,6 +1054,7 @@ end = struct
               | {pof_desc= Otag (_, t1); _} -> typ == t1
               | {pof_desc= Oinherit t1; _} -> typ == t1 ) )
       | Ptyp_class (_, l) -> assert (List.exists l ~f)
+      | Ptyp_of_kind _ -> assert false
```

Comment:
Should this look through the with-bounds? I know this construct will be rejected by the type-checker with with-bounds, so maybe it's not necessary.